backend/infrahub/core/diff/coordinator.py

```diff
@@ -147,13 +157,57 @@ async def create_or_update_arbitrary_timeframe_diff(
         )
         async with self.lock_registry.get(name=general_lock_name, namespace=self.lock_namespace):
             log.debug(f"Acquired lock to run arbitrary diff update for {base_branch.name} - {diff_branch.name}")
-            return await self._update_diffs(
+            base_enriched_diff, branch_enriched_diff = await self._update_diffs(
                 base_branch=base_branch,
                 diff_branch=diff_branch,
                 from_time=from_time,
                 to_time=to_time,
                 tracking_id=tracking_id,
             )
+            await self.summary_counts_enricher.enrich(enriched_diff_root=base_enriched_diff)
+            await self.diff_repo.save(enriched_diff=base_enriched_diff)
+            await self.summary_counts_enricher.enrich(enriched_diff_root=branch_enriched_diff)
+            await self.diff_repo.save(enriched_diff=branch_enriched_diff)
+            await self._update_core_data_checks(enriched_diff=branch_enriched_diff)
+            log.debug(f"Arbitrary diff update complete for {base_branch.name} - {diff_branch.name}")
+        return branch_enriched_diff
+
+    async def recalculate(
+        self,
+        base_branch: Branch,
+        diff_branch: Branch,
+        diff_id: str,
+    ) -> EnrichedDiffRoot:
+        general_lock_name = self._get_lock_name(
+            base_branch_name=base_branch.name, diff_branch_name=diff_branch.name, is_incremental=False
+        )
+        async with self.lock_registry.get(name=general_lock_name, namespace=self.lock_namespace):
+            log.debug(f"Acquired lock to recalculate diff for {base_branch.name} - {diff_branch.name}")
+            current_branch_diff = await self.diff_repo.get_one(diff_branch_name=diff_branch.name, diff_id=diff_id)
+            if current_branch_diff.tracking_id and isinstance(current_branch_diff.tracking_id, BranchTrackingId):
```

Comment:
there are different subclasses of `TrackingId` and we need to make sure that this is a `BranchTrackingId` instance (this is the current method of checking if a diff is for tracking a particular branch and it could be done more pythonically)